metadata-ingestion/src/datahub/ingestion/source/sac/sac.py

```diff
@@ -0,0 +1,808 @@
+                    env = connection.env
+                else:
+                    platform = model.system_type.lower()
+                    platform_instance = model.connection_id
+                    env = DEFAULT_ENV
+
+                upstream_dataset_urn = make_dataset_urn_with_platform_instance(
+                    platform=platform,
+                    name=upstream_dataset_name,
+                    platform_instance=platform_instance,
+                    env=env,
+                )
+
+                if upstream_dataset_urn not in self.ingested_upstream_dataset_keys:
+                    mcp = MetadataChangeProposalWrapper(
+                        entityType="dataset",
+                        changeType=ChangeTypeClass.UPSERT,
+                        entityUrn=upstream_dataset_urn,
+                        aspectName="datasetKey",
+                        aspect=dataset_urn_to_key(upstream_dataset_urn),
+                    )
+
+                    yield MetadataWorkUnit(
+                        id=f"dataset-key-{upstream_dataset_urn}",
+                        mcp=mcp,
+                        is_primary_source=False,
+                    )
+
+                    self.ingested_upstream_dataset_keys.add(upstream_dataset_urn)
+
+                mcp = MetadataChangeProposalWrapper(
+                    entityType="dataset",
+                    changeType=ChangeTypeClass.UPSERT,
+                    entityUrn=dataset_urn,
+                    aspectName="upstreamLineage",
+                    aspect=UpstreamLineageClass(
+                        upstreams=[
+                            UpstreamClass(
+                                dataset=upstream_dataset_urn,
+                                type=DatasetLineageTypeClass.COPY,
+                            ),
+                        ],
+                    ),
+                )
+
+                yield MetadataWorkUnit(
+                    id=f"dataset-upstream-lineage-{dataset_urn}", mcp=mcp
+                )
+            else:
+                logger.warning(
+                    f"Unknown upstream dataset for model with id {model.namespace}:{model.model_id} and external id {model.external_id}"
+                )
+                self.report.report_warning(
+                    "unknown-upstream-dataset",
+                    f"Unknown upstream dataset for model with id {model.namespace}:{model.model_id} and external id {model.external_id}",
+                )
+        elif model.system_type is not None:
+            logger.warning(
+                f"Unknown system type {model.system_type} for model with id {model.namespace}:{model.model_id} and external id {model.external_id}"
+            )
+            self.report.report_warning(
+                "unknown-system-type",
+                f"Unknown system type {model.system_type} for model with id {model.namespace}:{model.model_id} and external id {model.external_id}",
+            )
```

Comment:
`.warning` will also automatically log the message, so you can remove the logger.warning call

```suggestion
            self.report.warning(
                title="Cannot generate upstream lineage",
                message="Unknown system type - lineage will be incomplete.",
                context=f"system type {model.system_type} for model with id {model.namespace}:{model.model_id} and external id {model.external_id}",
            )
```

Reply:
There were probably some API changes in the past I missed (i.e. #10828), but also omiting everything except entityUrn and aspect in MetadataChangeProposalWrapper or the mcp.as_workunit method...;-)